hutch_python/tests/test_utils.py

```diff
@@ -119,3 +127,39 @@ def test_hutch_banner():
     logger.debug('test_hutch_banner')
     utils.hutch_banner()
     utils.hutch_banner('mfx')
+
+
+@pytest.fixture(scope='function')
+def RE_abort():
+    loop = asyncio.new_event_loop()
+    loop.set_debug(True)
+    RE = RunEngine({}, loop=loop,
+                   context_managers=[utils.AbortSigintHandler])
+
+    yield RE
+
+    if RE.state != 'idle':
+        RE.halt()
+
+
+def test_abort_RE(RE_abort):
```

Comment:
Tiny detail here, `SIGINT` tests using threaded `os.kill` calls don't work on windows (they terminate the entire test suite...)
https://docs.python.org/3/library/os.html#os.kill
I don't actually know any other ways to test this, I've been skipping these to move on with my life:
```suggestion
@skip_if_win32_generic
def test_abort_RE(RE_abort):
```

Reply:
Naive guess since I haven't looked into details of this since bluesky-2015, but I'd imagine `raise KeyboardInterrupt` in the plan wouldn't be the same, would it?

Reply:
I should have documented my previous efforts better. Last time I tried `raise KeyboardInterrupt` and I also tried using the alternate windows-only signal and had some sort of problem each time.

Reply:
I was looking around and saw something about mapping a windows Ctrl+C signal to sigint before hitting the handler, which might be a good way forward.  Do we want to try to support windows proper?  Our hutch environments are all on linux right?

Reply:
We've been more or less ignoring proper support for Windows, but it's nice for the tests to not break entirely when you're developing something else in the library on a Windows machine.